index.d.ts

```diff
@@ -11,28 +15,31 @@ declare class BeeQueue {
 
   on(ev: "ready",     fn: () => void): this;
   on(ev: "error",     fn: (err: Error) => void): this;
-  on(ev: "succeeded", fn: (job: BeeQueue.Job, result: any) => void): this;
-  on(ev: "retrying",  fn: (job: BeeQueue.Job, err: Error) => void): this;
-  on(ev: "failed",    fn: (job: BeeQueue.Job, err: Error) => void): this;
+  on(ev: "succeeded", fn: (job: BeeQueue.Job<T>, result: any) => void): this;
+  on(ev: "retrying",  fn: (job: BeeQueue.Job<T>, err: Error) => void): this;
+  on(ev: "failed",    fn: (job: BeeQueue.Job<T>, err: Error) => void): this;
   on(ev: "stalled",   fn: (jobId: string) => void): this;
 
   on(ev: "job succeeded", fn: (jobId: string, result: any) => void): this;
   on(ev: "job retrying",  fn: (jobId: string, err: Error) => void): this;
   on(ev: "job failed",    fn: (jobId: string, err: Error) => void): this;
   on(ev: "job progress",  fn: (jobId: string, progress: number) => void): this;
 
-  createJob<T>(data: T): BeeQueue.Job;
+  ready(): Promise<this>;
+  ready(cb: (err: Error | null) => void): this;
```

Comment:
The callback here is optional and Promise<this> is always returned.

Reply:
Yes, you're right @calebboyd Fixed that.